bazel/external/lightstep.BUILD

```diff
@@ -0,0 +1,73 @@
+load("@//bazel/external:helpers.bzl", "cc_genrule_deps")
+
+cc_library(
+    name = "lightstep",
+    srcs = [":_prefix/lib/liblightstep_core_cxx11.a"],
+    hdrs = glob([
+        "src/c++11/lightstep/**/*.h",
+        "src/c++11/mapbox_variant/**/*.hpp",
+    ]) + [
+        ":_prefix/include/collector.pb.h",
+        ":_prefix/include/lightstep_carrier.pb.h",
+    ],
+    includes = ["src/c++11", "_prefix/include"],
+    deps = ["@protobuf_bzl//:protobuf"],
+    visibility = ["//visibility:public"],
+)
+
+cc_genrule_deps(
+    name = "protobuf_deps",
+    deps = ["@protobuf_bzl//:protobuf"],
+)
+
+genrule(
+    name = "install",
+    srcs = glob(["**"]) + [
+        ":protobuf_deps",
+        "@protobuf_bzl//:well_known_protos",
+    ],
+    cmd = """
+    configure="$${PWD}/$(location :configure)"
```

Comment:
Not with `new_http_archive`, but using a smarter repository definer should let us put in additional files during the "fetch" step. I'll try to get that figured out.

Reply:
I would caution against adding repository rules that perform build, or you might hit the issue that led to the single massive recursive make repository rule in the current implementation. Repository rules execute sequentially, so you don't get any cross dependency build parallelism. This slowed down the initial fetch significantly, e.g. from a couple of minutes to tens of minutes.

Reply:
Well, that's what we do today https://github.com/envoyproxy/envoy/blob/master/bazel/repositories.bzl#L12. I'm starting to wonder how the new approach (if we're moving to repository rules for everything) is moving away from the existing approach, other than decomposing the monolith recursive make, which provide parallelism via job control, to a bunch of repository rules that will be executed sequentially with no job control.

Reply:
The future repository rules should be very fast (just unpacking a tarball, maybe writing a shell blob). I want all the real build work to happen within standard BUILD rules, so they can be parallelized.

Reply:
That could work. I'm still a little unsure on how this will work though with output file determination, since output files may only be known following the build process (even headers might be generated), so figuring out the output artifacts is going to require a post-build step.